set.mm

```diff
@@ -520332,6 +520762,300 @@ such that every nonzero prime ideal contains a prime element (this is a
+           (Contributed by Thierry Arnoux, 3-Jun-2025.) $)
+        1arithufdlem2 $p |- ( ph -> ( X .x. Y ) e. S ) $=
+          ( vg vh co cgsu wceq cword wrex crab eqeq1 ufdcringd crngringd ssrab3
+          cv rexbidv sselid ringcld wcel wa cconcat oveq2 eqeq2d ccatcl simpllr
+          ad5ant24 simpr oveq12d cmnd crg ringmgp syl ad4antr wss adantr rprmcl
+          ex ssrdv sswrd simp-4r sseldd simplr mgpbas mgpplusg gsumccat syl3anc
+          cufd eqtr4d rspcedvdw eleqtrdi cbvrexvw bitrid elrab3 biimpa ad2antrr
+          syl2anc r19.29a elrabd eleqtrrdi ) AKLGUHZBURZJIURZUIUHZUJZIDUKZULZBC
+          UMZFAXIXCXFUJZIXHULZBXCCXDXCUJXGXKIXHXDXCXFUNUSACEGKLNUCAEAESUOUPZAFC
+          KXIBCFUBUQZUDUTZAFCLXNUEUTZVAAKJUFURZUIUHZUJZXLUFXHAXQXHVBZVCZXSVCZLJ
+          UGURZUIUHZUJZXLUGXHYBYCXHVBZVCZYEVCZXKXCJXQYCVDUHZUIUHZUJIYIXHXEYIUJX
+          FYJXCXEYIJUIVEVFXTYFYIXHVBAXSYEDXQYCVGVIYHXCXRYDGUHZYJYHKXRLYDGYAXSYF
+          YEVHYGYEVJVKYHJVLVBZXQCUKZVBYCYMVBYJYKUJAYLXTXSYFYEAEVMVBYLXMEJRVNVOV
+          PYHXHYMXQAXHYMVQZXTXSYFYEADCVQYNABDCAXDDVBZXDCVBAYOVCCDEWJXDNQAEWJVBY
+          OSVRAYOVJVSVTWADCWBVOVPZAXTXSYFYEWCWDYHXHYMYCYPYBYFYEWEWDCGJXQYCCEJRN
+          WFEGJRUCWGWHWIWKWLAYEUGXHULZXTXSALCVBZLXJVBZYQXPALFXJUEUBWMYRYSYQXIYQ
+          BLCXIXDYDUJZUGXHULXDLUJZYQXGYTIUGXHXEYCUJXFYDXDXEYCJUIVEVFWNUUAYTYEUG
+          XHXDLYDUNUSWOWPWQWSWRWTAKCVBZKXJVBZXSUFXHULZXOAKFXJUDUBWMUUBUUCUUDXIU
+          UDBKCXIXDXRUJZUFXHULXDKUJZUUDXGUUEIUFXHXEXQUJXFXRXDXEXQJUIVEVFWNUUFUU
+          EXSUFXHXDKXRUNUSWOWPWQWSWTXAUBXB $.
+      $}
+
+      ${
+        $d .0. c d f p y z $.  $d .0. d f k p t w x y z $.
+        $d .0. d k p t v w y z $.  $d .x. c d p y z $.
+        $d .x. d f k p t w x y z $.  $d .x. v $.  $d B c d p y z $.
+        $d B d f k p t w x y z $.  $d B v $.  $d K u y $.  $d M a b $.
+        $d M c d p y z $.  $d M d f k p t w x y z $.  $d M v $.  $d P a $.
+        $d P c d p y z $.  $d P d f k p t w x y z $.  $d P i $.  $d P v $.
+        $d R f k p x y $.  $d R i j $.  $d S a b $.  $d S c d p y z $.
+        $d S d f k p t w x y z $.  $d S i j $.  $d S u $.  $d S v $.
+        $d U c d p y z $.  $d U d f k p t w x y z $.  $d U v $.
+        $d X c p y z $.  $d X f p x y z $.  $d X i j $.  $d X u $.
+        $d Y f x y $.  $d a b f x $.  $d a b ph $.  $d c d p ph y z $.
+        $d d f k p ph t w x y z $.  $d i j p $.  $d i j ph $.  $d ph u $.
+        $d ph v $.  $d ps u $.  $d ps y $.
+        1arithufdlem.3 $e |- ( ph -> X e. B ) $.
+        1arithufdlem.4 $e |- ( ph -> -. X e. U ) $.
+        1arithufdlem.5 $e |- ( ph -> X =/= .0. ) $.
+        ${
+          1arithufdlem3.p $e |- .x. = ( .r ` R ) $.
+          1arithufdlem3.y $e |- ( ph -> Y e. B ) $.
+          1arithufdlem3.1 $e |- ( ph -> ( Y .x. X ) e. S ) $.
+          $( Lemma for ~ 1arithufd .  If a product ` ( Y .x. X ) ` can be
+             written as a product of primes, with ` X ` non-unit, nonzero, so
```

Comment:
This is a very minor nitpick which can be safely ignored, but it's better to choose whether we hyphenate non- or not. I.e. write "nonunit, nonzero" or "non-unit, non-zero".

Reply:
In https://github.com/metamath/set.mm/pull/4872/commits/cc2150ead40d9ee3851e56d8f37942b85b152b14, I chose to use "non-unit, non-zero", however I'm open to any suggestion.